sqlparser/lib/src/analysis/steps/prepare_ast.dart

```diff
@@ -249,4 +255,18 @@ class AstPreparingVisitor extends RecursiveVisitor<void, void> {
 
     visitChildren(e, null);
   }
+
+  /// If a nested query was found. Collect everything separately.
+  @override
+  void visitMoorSpecificNode(MoorSpecificNode e, void arg) {
+    if (e is NestedQueryColumn) {
+      // create a new scope for the nested query to differentiate between
+      // references that can be resolved in the nested query and references
+      // which require data from the parent query
+      e.select.scope = e.scope.createChild();
+      AstPreparingVisitor(context: context).start(e.select);
+    } else {
+      super.visitMoorSpecificNode(e, arg);
+    }
+  }
```

Comment:
This is also the reason why a new `AstPreparingVisitor` is created for every subquery.